vyper/venom/passes/memmerging.py

```diff
@@ -182,15 +189,28 @@ def _barrier():
 
         self._optimize_copy(bb, intervals, copy_inst)
 
+    def _add_zero_interval(self, intervals: list[_Interval], new_inter: _Interval) -> bool:
+        index = bisect_left(intervals, new_inter)
+        intervals.insert(index, new_inter)
+
+        i = max(index - 1, 0)
+        while i < min(index + 1, len(intervals) - 1):
+            merged = intervals[i].inclusive_merge(intervals[i + 1])
+            if merged:
+                del intervals[i + 1]
+            else:
+                i += 1
```

Comment:
correct me if i'm wrong but here we can loop all the way to the end right? but due to the way `intervals` was set up, we can only ever touch two intervals (the ones before and after, as in overlap_exist`)?

Reply:
due to how the indexes are set up here it should touch only intervals that are around the inserted interval.

Reply:
ah, i misread the bound